### PR TITLE
replace old infra with @lume/cli, convert to TypeScript

### PR DESCRIPTION
**Pull request by trusktr**
Hello!

You may not want to merge this, as I'm converted to TypeScript and using my own tooling (`@lume/cli`, which abstracts TS, Webpack, karma test runner, etc), but it may serve to get ideas spinning.

Main reason I did this is because it makes the codebase output compatible with all ESM environments and tooling. I looked at testee and pika, but pika is no longer a thing, so I decided to just use what I know:

- [x] use `@lume/cli` (it uses typescript for ESM output and webpack for global script output, with karma-electron for unit tests)
- [x] convert code to TypeScript
- [ ] convert unit tests to lume/cli compatible

The output in `dist/` is the TS files converted JS (ESM), and also a `dist/global.js` file that `<script>` tag users can use to "polyfill" it onto global without using ESM.

I'm going through my [LUME project](https://github.com/lume/lume/) and making sure all packages (and their dependencies) work properly in

- unpkg.com with `?module` for ESM mode
- Skypack
- native browser ESM
- node ESM
- Webpack (which follows Node ESM)
- Meteor (has its own ESM to CommonJS system and runtime is still CommonJS).
- Rollup
Basically make sure all parts of my dep graph are ESM that works everywhere. At the moment, importing my stuff in some places fails due to custom-attributes and other dependencies not being up to date for latest ESM stuff yet.

### CODE REVIEW

**Comment by trusktr** [comment]
It works on everywhere!

F.e. try importing `element-behaviors` which imports `@lume/custom-attributes` in console from unpkg.com:

```js
import('https://unpkg.com/element-behaviors@2.2.9?module').then(m => {
// or import('https://unpkg.com/element-behaviors@2.2.9/dist/index.js?module').then(m => {
  console.log(m)
  console.log(elementBehaviors)
  console.log(customAttributes)
})
```

Or try `@lume/custom-attributes` directly:

```js
import('https://unpkg.com/@lume/custom-attributes@0.1.2?module').then(m => {
// or import('https://unpkg.com/@lume/custom-attributes@0.1.2/dist/index.js?module').then(m => {
  console.log(m)
  console.log(customAttributes)
})
```

Make sure to try it on a site without CORS restricting it.

Skypack:

```js
import('https://cdn.skypack.dev/element-behaviors@2.2.9').then(m => {
  console.log(m)
  console.log(elementBehaviors)
  console.log(customAttributes)
})
```

or

```js
import('https://cdn.skypack.dev/@lume/custom-attributes@0.1.2').then(m => {
  console.log(m)
  console.log(customAttributes)
})
```

And JSPM (_so faaaast, even on first time ever loading a lib!_ :car: :dash:):

```js
import('https://jspm.dev/element-behaviors@2.2.9').then(m => {
  console.log(m)
  console.log(elementBehaviors)
  console.log(customAttributes)
})
```

or

```js
import('https://jspm.dev/@lume/custom-attributes@0.1.2').then(m => {
  console.log(m)
  console.log(customAttributes)
})
```

**Comment by matthewp** [comment]
This is really cool!  But yeah I don't want such big infrastructure in this project. I think a fork is a good idea, as you seem more active on this front that I've been lately. Wish you the best of luck here!
